go/shared.go

```diff
@@ -1,3 +1,38 @@
 package app
 
+import (
+	"log"
+
+	"go.temporal.io/sdk/workflow"
+)
+
 const BreakfastTaskQueue = "BREAKFAST_TASK_QUEUE"
+
+type Compensations []any
+
+func (s *Compensations) AddCompensation(activity any) {
+	*s = append(*s, activity)
+}
+
+func (s Compensations) Compensate(ctx workflow.Context, in_parallel bool) {
+	if !in_parallel {
+		for i := len(s) - 1; i >= 0; i-- {
+			errCompensation := workflow.ExecuteActivity(ctx, s[i]).Get(ctx, nil)
+			if errCompensation != nil {
+				log.Println("Executing compensation failed", errCompensation)
```

Comment:
Should not use stdlib log inside of workflows. Consider `workflow.GetLogger`.

Reply:
oh, got it! Thanks!